drivers/i2s/i2s_ll_stm32.c

```diff
@@ -0,0 +1,1152 @@
+
+	active_dma_tx_channel[stream->dma_channel] = NULL;
+}
+
+static void rx_queue_drop(struct stream *stream)
+{
+	size_t size;
+	void *mem_block;
+
+	while (queue_get(&stream->mem_block_queue, &mem_block, &size) == 0) {
+		k_mem_slab_free(stream->cfg.mem_slab, &mem_block);
+	}
+
+	k_sem_reset(&stream->sem);
+}
+
+static void tx_queue_drop(struct stream *stream)
+{
+	size_t size;
+	void *mem_block;
+	unsigned int n = 0;
+
+	while (queue_get(&stream->mem_block_queue, &mem_block, &size) == 0) {
+		k_mem_slab_free(stream->cfg.mem_slab, &mem_block);
+		n++;
+	}
+
+	for (; n > 0; n--) {
+		k_sem_give(&stream->sem);
+	}
+}
+
+static struct device *get_dev_from_rx_dma_channel(u32_t dma_channel)
+{
+	return active_dma_rx_channel[dma_channel];
+}
+
+static struct device *get_dev_from_tx_dma_channel(u32_t dma_channel)
+{
+	return active_dma_tx_channel[dma_channel];
+}
+
+#ifdef CONFIG_I2S_1
+static struct device DEVICE_NAME_GET(i2s_stm32_1);
+
+static void i2s_stm32_irq_config_func_1(struct device *dev);
+
+static const struct i2s_stm32_cfg i2s_stm32_config_1 = {
+	.i2s = (SPI_TypeDef *) CONFIG_I2S_1_BASE_ADDRESS,
+	.pclken = {
+		.enr = LL_APB2_GRP1_PERIPH_SPI1,
+		.bus = STM32_CLOCK_BUS_APB2,
+	},
+	.i2s_clk_sel = CLK_SEL_2,
+	.irq_config = i2s_stm32_irq_config_func_1,
+};
+
+struct queue_item rx_1_ring_buf[CONFIG_I2S_STM32_RX_BLOCK_COUNT + 1];
+struct queue_item tx_1_ring_buf[CONFIG_I2S_STM32_TX_BLOCK_COUNT + 1];
+
+static struct i2s_stm32_data i2s_stm32_data_1 = {
+	.dma_name = I2S1_DMA_NAME,
+	.rx = {
+		.dma_channel = I2S1_DMA_CHAN_RX,
+		.dma_cfg = {
+			.block_count = 1,
+			.dma_slot = I2S1_DMA_SLOT_RX,
+			.channel_direction = PERIPHERAL_TO_MEMORY,
+			.source_data_size = 1,  /* 16bit default */
+			.dest_data_size = 1,    /* 16bit default */
+			.source_burst_length = 0, /* SINGLE transfer */
+			.dest_burst_length = 1,
+			.dma_callback = dma_rx_callback,
+		},
+		.stream_start = rx_stream_start,
+		.stream_disable = rx_stream_disable,
+		.queue_drop = rx_queue_drop,
+		.mem_block_queue.buf = rx_1_ring_buf,
+		.mem_block_queue.len = ARRAY_SIZE(rx_1_ring_buf),
+	},
+	.tx = {
+		.dma_channel = I2S1_DMA_CHAN_TX,
+		.dma_cfg = {
+			.block_count = 1,
+			.dma_slot = I2S1_DMA_SLOT_TX,
+			.channel_direction = MEMORY_TO_PERIPHERAL,
+			.source_data_size = 1,  /* 16bit default */
+			.dest_data_size = 1,    /* 16bit default */
+			.source_burst_length = 1,
+			.dest_burst_length = 0, /* SINGLE transfer */
+			.dma_callback = dma_tx_callback,
+		},
+		.stream_start = tx_stream_start,
+		.stream_disable = tx_stream_disable,
+		.queue_drop = tx_queue_drop,
+		.mem_block_queue.buf = tx_1_ring_buf,
+		.mem_block_queue.len = ARRAY_SIZE(tx_1_ring_buf),
+	},
+};
+DEVICE_AND_API_INIT(i2s_stm32_1, CONFIG_I2S_1_NAME, &i2s_stm32_initialize,
+		    &i2s_stm32_data_1, &i2s_stm32_config_1, POST_KERNEL,
+		    CONFIG_I2S_INIT_PRIORITY, &i2s_stm32_driver_api);
+
+static void i2s_stm32_irq_config_func_1(struct device *dev)
+{
+	SYS_LOG_DBG("Enable I2S1 IRQ");
```

Comment:
Can you clean up this dgb message?
Not sure it adds a real value.

Reply:
ack